CMakeLists.txt

```diff
@@ -82,15 +82,23 @@ ExternalProject_Add(opencv3_src
     -DBUILD_opencv_xfeatures2d=ON
     -DBUILD_opencv_ximgproc=OFF
     -DBUILD_opencv_xobjdetect=OFF
-    -DBUILD_opencv_xphoto=OFF .
-  BUILD_COMMAND cd ../${OPENCV_SRC_PATH} && make -j8
-  INSTALL_COMMAND cd ../${OPENCV_SRC_PATH} && make install -j8
+    -DBUILD_opencv_xphoto=OFF ../${OPENCV_SRC_PATH}
+  BUILD_COMMAND cd ../${OPENCV_BUILD_PATH} && make -j8
+  INSTALL_COMMAND cd ../${OPENCV_BUILD_PATH} && make install -j8
 )
 
+
 cs_add_library(${PROJECT_NAME} src/dependency_tracker.cc)
 add_dependencies(${PROJECT_NAME} opencv3_src)
-target_link_libraries(${PROJECT_NAME}
-  ${CATKIN_DEVEL_PREFIX}/lib/libopencv_core${CMAKE_SHARED_LIBRARY_SUFFIX})
+# This can be removed once Ubuntu 14.04 is no longer supported.
+execute_process(COMMAND lsb_release -r OUTPUT_VARIABLE UBUNTU_RELEASE)
+if("${UBUNTU_RELEASE}" MATCHES "14.04")
+  target_link_libraries(${PROJECT_NAME}
+    ${CATKIN_DEVEL_PREFIX}/lib/x86_64-linux-gnu/libopencv_core${CMAKE_SHARED_LIBRARY_SUFFIX})
+else()
+  target_link_libraries(${PROJECT_NAME}
```

Comment:
what is this library used for?

Reply:
I think this library is needed and as a matter of fact, it will be removed per #17